src/Workspaces/Core/Portable/Shared/Extensions/ISymbolExtensions.cs

```diff
@@ -865,6 +865,12 @@ public static ITypeSymbol GetSymbolType(this ISymbol symbol)
                 return aliasSymbol.Target as ITypeSymbol;
             }
 
+            var methodSymbol = symbol as IMethodSymbol;
```

Comment:
No.  The idea is that the caller of this method needs ot handle IMethodSymbol's appropriately.  it's not always correct to say that the type of the method is it's return type. 

That's appropriate if we have a method *invocation*.  i.e.  the type of "Foo()" is whatever the return type of "Foo" is.  But the type of "Foo" itself is unclear.  For example, we have features that view the type of "Foo" to be the Func/Actoin that best describes its signature.

Who was calling this that you needed this specialized behavior for?  We should be updating that code instead.